backend/core/models/user.py

```diff
@@ -227,21 +227,78 @@ def trusted_users(cls) -> QuerySet["User"]:
     def supertrusted_seed_users(cls) -> QuerySet["User"]:
         return cls.trusted_users().filter(is_staff=True)
 
+    @classmethod
+    def validate_email_unique_with_plus(cls, email: str, username="") -> str:
+        """Raise ValidationError when similar emails are found in the database.
+
+        Keyword arguments:
+        email -- An email that is going to be written in the database.
+        username -- The logged user's username, used to exclude him/herself from
+                    the validation when updating its own email. Empty for
+                    anonymous users.
+
+        Emails considered similar when:
+            - they share the same non-case-sensitive domain ;
+            - they share, in the local part, the same non case-sensitive string
+              before the `+` symbol.
+
+        Examples of emails considered similar:
+            - bob@example.org
+            - bob+@example.org
+            - bob+tournesol@example.org
+            - BOB+tournesol@example.org
+            - bob+hello@example.org
+            - BOB+HELLO@example.org
+            - etc.
+        """
+        email_split = email.split("@")
+
+        # if there is no `@`, do nothing
+        if len(email_split) == 1:
+            return email
+
+        local_part = email_split[0]
```

Comment:
Theoretically the local part may contain one or multiple `@`. So `.rsplit("@", 1)` should be used instead to get the local part.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/825/commits/d3e35e798b2827a6a37902a4ee837f4a1d4f9ac8.